sys/mips/pmap.c

```diff
@@ -307,6 +308,31 @@ static bool pmap_extract_nolock(pmap_t *pmap, vaddr_t va, paddr_t *pap) {
   return true;
 }
 
+bool pmap_extract_and_hold(pmap_t *pmap, vaddr_t va, paddr_t *pap,
+                           vm_prot_t prot) {
+  assert(pap != NULL);
+  paddr_t old = *pap;
+
+  SCOPED_MTX_LOCK(&pmap->mtx);
+
+  bool result = pmap_extract_nolock(pmap, va, pap);
+
+  if (!result) {
+    *pap = old;
+    return false;
+  }
```

Comment:
You don't need `result` here. You could call `pmap_extract_nolock` directly in `if` statement.

Reply:
Changed.